group-coordinator/src/main/java/org/apache/kafka/coordinator/group/assignor/GeneralUniformAssignmentBuilder.java

```diff
@@ -14,17 +14,903 @@
+     * </ul>
+     *
+     * @param partition         The partition to reassign.
+     * @throws AssertionError   If no subscribed member is found.
+     */
+    private void reassignPartition(TopicIdPartition partition) {
+        // Find the new member with the least assignment size.
+        String newOwner = null;
+        for (String anotherMember : sortedMembersByAssignmentSize) {
+            if (members.get(anotherMember).subscribedTopicIds().contains(partition.topicId())) {
+                newOwner = anotherMember;
+                break;
+            }
+        }
+
+        if (newOwner == null) {
+            throw new PartitionAssignorException("No suitable new owner was found for the partition" + partition);
+        }
+
+        reassignPartition(partition, newOwner);
+    }
+
+    /**
+     * Reassigns the given partition to a new member while considering partition movements and stickiness.
+     * <p>
+     * This method performs the following actions:
+     * <ol>
+     *   <li> Determines the current owner of the partition. </li>
+     *   <li> Identifies the correct partition to move, adhering to stickiness constraints. </li>
+     *   <li> Processes the partition movement to the new member. </li>
+     * </ol>
+     *
+     * @param partition     The {@link TopicIdPartition} to be reassigned.
+     * @param newMember     The Id of the member to which the partition should be reassigned.
+     */
+    private void reassignPartition(TopicIdPartition partition, String newMember) {
+        String member = partitionOwnerInTargetAssignment.get(partition);
+        // Find the correct partition movement considering the stickiness requirement.
+        TopicIdPartition partitionToBeMoved = partitionMovements.getTheActualPartitionToBeMoved(
+            partition,
+            member,
+            newMember
+        );
+        processPartitionMovement(partitionToBeMoved, newMember);
+    }
+
+    private void processPartitionMovement(TopicIdPartition topicIdPartition, String newMember) {
+        String oldMember = partitionOwnerInTargetAssignment.get(topicIdPartition);
+
+        partitionMovements.movePartition(topicIdPartition, oldMember, newMember);
+
+        assignmentManager.removePartitionFromTargetAssignment(topicIdPartition, oldMember);
+        assignmentManager.addPartitionToTargetAssignment(topicIdPartition, newMember);
+    }
+
+    /**
+     * <code> MemberPair </code> represents a pair of member Ids involved in a partition reassignment. Each
+     * <code> MemberPair </code> object, which contains a source (<code>src</code>) and a destination (<code>dst</code>)
+     * element, normally corresponds to a particular partition or topic, and indicates that the particular partition or some
+     * partition of the particular topic was moved from the source member to the destination member during the rebalance.
+     */
+    private static class MemberPair {
```

Comment:
I understand your concern but since MemberPair is always used with src and dest I feel like it should be easy enough to comprehend esp with the java docs but if it seems too confusing I can change it. What do you think?